apps/prairielearn/src/lib/url.ts

```diff
@@ -6,3 +6,7 @@ export function getCanonicalHost(req: Request): string {
   if (config.serverCanonicalHost) return config.serverCanonicalHost;
   return `${req.protocol}://${req.get('host')}`;
 }
+
+export function getSearchParams(req: Request): URLSearchParams {
+  return new URL(req.originalUrl, getCanonicalHost(req)).searchParams;
+}
```

Comment:
It took me _just_ long enough to remember how to do this that I figured I'd just make this little library function.